rotkehlchen/chain/ethereum/modules/aave/v2/decoder.py

```diff
@@ -74,9 +78,11 @@ def _decode_deposit(
     ) -> None:
         """Decode aave v2 deposit event"""
         user = hex_or_bytes_to_address(tx_log.data[:32])
-        on_behalf_of = hex_or_bytes_to_address(tx_log.data[64:96]) if hex_or_bytes_to_str(tx_log.data[64:96]) != '' else None  # noqa: E501
+        on_behalf_of = hex_or_bytes_to_address(tx_log.topics[2]) if hex_or_bytes_to_str(tx_log.topics[2]) != '' else None  # noqa: E501
```

Comment:
Why did you change this? Was it wrong before?

Reply:
yes, I have checked and there is no third element in the data, there are 3 topics and in the data only one address and one amount